docs/6-infrastructure-configuration-management/6.1.2-terraform-backends.md

```diff
@@ -0,0 +1,67 @@
+
+If you ran through the AWS tutorial use the [S3 backend](https://developer.hashicorp.com/terraform/language/settings/backends/s3) for this exercise. For the Azure tutorial use the [azurerm backend](https://developer.hashicorp.com/terraform/language/settings/backends/azurerm).
+
+!> Terraform has not implemented state migration from Terraform Cloud to another backend so you will need to clean up the previous state with the following steps
+
+1. Run `terraform destroy` to remove existing resources.
+
+2. Delete the `.terraform` folder.
+
+3. Create cloud resources to store the Terraform state in. The specific resources you need will depend on the cloud platform and backend you are using. Read the documentation for the specific backend and Terraform resources for more information. 
+
+4. Remove the `cloud` configuration block from the `terraform` block of your Terraform configuration.
+
+5. Add a `backend` configuration block to the `terraform` block for your backend. The backend configuration should reference the resources you created in step #3.
+
+6. Run `terraform init` and `terraform apply` to initialize the new backend and apply the configuration.
+
+7. Look at the cloud storage you setup for the Terraform backend. What files have been created and what do they contain?
+
+8. Push up you changes and have a teammate pull down you changes and apply a change
+
+?> They may need additional access to the resources you've created depending on the platform you are using and what default permission they have.
+
+**Deep Dive**
+
+- What is stored in the Terraform state?
+
+- What are some pros/cons of using Terraform Cloud to store Terraform state vs another backend?
+
+**Automation**
+
+Sharing a Terraform workspace is important for teams to be able to manage infrastructure. This can also cause problems if different members of the team apply changes before they are merged into a main branch in source control. One way to avoid this is to create CI/CD pipelines to test and apply change and enforce that the main branch is your source of truth for the state of the Terraform.
+
+1. Create a GitHub workflow for your repo that triggers when an pull request to the main branch is created or updated that runs `terraform fmt`, `terraform init`, `terraform validate` and `terraform plan`.
+
+?> You will need to configuring the GitHub runner environment so it can authenticate with the cloud provider and has the correct permission
+
+!> Do not commit authentication credentials to your source repository
+
+2. Ensure that the pipeline triggers when you create a PR and that all of your Terraform checks work as expected.
+
+3. Create a second workflow that is triggered when a commit is pushed to the main branch that runs `terraform init` and `terraform apply`
+
+4. Ensure that merging changes into your main branch trigger the workflow and that changes are applied to your Terraform state.
```

Comment:
"Ensure that merging changes into your main branch [will] trigger the workflow and that changes are applied to your Terraform state."